vault/mount.go

```diff
@@ -365,6 +377,7 @@ func (c *Core) remount(src, dst string) error {
 		return err
 	}
 
+	c.mountsLock.Lock()
```

Comment:
Here it looks like `Entries` is also protected, prompting the need to use a read lock in `Tainted()`.
If not, can this lock acquisition be placed just above `persistMounts`?